### PR TITLE
Sample relations in Cuffnorm

### PR DESCRIPTION
**Pull request by jkokosar**
@hadalin fyi

### CODE REVIEW

File: resolwe_bio/expression_filters/relation.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+def replicate_groups(data):
+    """Get replicate groups."""
+    if not isinstance(data, list):
+        raise ValueError('List of data IDs must be given')
+
+    data_ids = [id_(d) for d in data]
+    samples = Sample.objects.filter(data__id__in=data_ids)
+
+    if len(samples) != len(data_ids):
+        raise ValueError('Can not get replicates for data without sample')
```
[the remark this answers is not in the document]
Comment (mstajdohar) [reply resolwe_bio/expression_filters/relation.py]: Good point, I'll fix it.

File: resolwe_bio/expression_filters/relation.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    partitions = RelationPartition.objects.filter(relation__category='Replicate', relation__type__name='group',
+                                                  entity__in=samples)
+
+    sample_group = {}
+    for p in partitions:
+        if p.entity.id in sample_group:
+            raise ValueError('Can not handle samples with more then one replicate relations')
```
[the remark this answers is not in the document]
Comment (mstajdohar) [reply resolwe_bio/expression_filters/relation.py]: Good point, I'll fix it.

File: resolwe_bio/expression_filters/relation.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from resolwe.flow.expression_engines.jinja.filters import id_
+from resolwe.flow.models.entity import RelationPartition
+
+from resolwe_bio.models import Sample
+
+
+def replicate_groups(data):
+    """Get replicate groups."""
+    if not isinstance(data, list):
+        raise ValueError('List of data IDs must be given')
```
Comment (jvrakor) [inline resolwe_bio/expression_filters/relation.py]: List of data, list of data IDs, or both?

File: resolwe_bio/expression_filters/relation.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    if len(data_ids) != len(set(data_ids)):
+        raise ValueError('Repeated data objects not allowed')
+
+    samples = Sample.objects.filter(data__id__in=data_ids)
+
+    if len(samples) != len(data_ids):
+        raise ValueError('Can not get replicates of data without sample')
+
+    partitions = RelationPartition.objects.filter(relation__category='Replicate',
```
Comment (jvrakor) [inline resolwe_bio/expression_filters/relation.py]: This format might be better: https://github.com/genialis/resolwe-bio/pull/657/files#diff-fbfd490016a4d3a3217974dcdf871ff9R98

**Comment by jkokosar** [comment]
@mstajdohar LGTM

**Pull request merged**